packages/client/hmi-client/src/services/document-assets.ts

```diff
@@ -171,11 +172,33 @@ async function getDocumentFileAsText(documentId: string, fileName: string): Prom
 	return response.data;
 }
 
+async function addDocumentFromDOI(
+	documentId: string,
+	doi: string,
+	filename: string
+): Promise<string | null> {
+	if (!documentId) return null;
+	const response = await API.put(
+		`/document-asset/${documentId}/uploadDocumentFromDOI?doi=${doi}&filename=${filename}`
+	);
+
+	if (!response || response.status >= 400) {
+		logger.error('Error upload file from doi');
+		return null;
+	}
+
+	// fire and forget pdf to cosmos
+	extractTextFromPDFDocument(documentId);
```

Comment:
This should be done in the `hmi-server` not here.

Reply:
moved to `hmi-server`